crates/astria-conductor/src/conductor.rs

```diff
@@ -94,6 +96,9 @@ impl Conductor {
     /// actors could not be spawned (executor, sequencer reader, or data availability reader).
     /// This usually happens if the actors failed to connect to their respective endpoints.
     pub fn new(cfg: Config) -> eyre::Result<Self> {
+        static METRICS: OnceLock<Metrics> = OnceLock::new();
```

Comment:
The intent here is to actively avoid creating a _global_ static, since that's what ties our hands during testing.

I agree the approach in this PR is fairly intrusive, but by passing a static ref down to the callsites we can instantiate multiple tests concurrently and provide each test instance with its own exclusive `Metrics` struct, avoiding munging the metrics between tests.

As an aside, I considered just passing the `Metrics` struct everywhere by value, cloning as required.  That's feasible with the implementation in this PR.  The individual metrics are all just wrappers around an `Arc`, so relatively cheap to clone.  But not as cheap as a static ref, so I opted for that (the Composer's `Metrics` is probably the most expensive to clone, holding five `HashMap`s of counters and another eight individual metrics).